app/src/main/java/de/danoeh/antennapod/config/StorageCallbacksImpl.java

```diff
@@ -149,5 +150,41 @@ public void onUpgrade(SQLiteDatabase db, int oldVersion, int newVersion) {
             db.execSQL(PodDBAdapter.CREATE_INDEX_QUEUE_FEEDITEM);
             db.execSQL(PodDBAdapter.CREATE_INDEX_SIMPLECHAPTERS_FEEDITEM);
         }
+        if(oldVersion <= 15) {
+            db.execSQL("ALTER TABLE " + PodDBAdapter.TABLE_NAME_FEED_MEDIA
+                    + " ADD COLUMN " + PodDBAdapter.KEY_HAS_EMBEDDED_PICTURE + " INTEGER DEFAULT -1");
+            Thread t = new Thread(new Runnable() {
+                public void run() {
+                    db.execSQL("UPDATE " + PodDBAdapter.TABLE_NAME_FEED_MEDIA
+                            + " SET " + PodDBAdapter.KEY_HAS_EMBEDDED_PICTURE + "=0"
+                            + " WHERE " + PodDBAdapter.KEY_DOWNLOADED + "=0");
+                    Cursor c = db.rawQuery("SELECT " + PodDBAdapter.KEY_FILE_URL + " FROM " + PodDBAdapter.TABLE_NAME_FEED_MEDIA +
+                            " WHERE " + PodDBAdapter.KEY_DOWNLOADED + "=1 AND " + PodDBAdapter.KEY_HAS_EMBEDDED_PICTURE + "=-1", null);
+                    if(c.moveToFirst()) {
+                        MediaMetadataRetriever mmr = new MediaMetadataRetriever();
+                        do {
+                            String fileUrl = c.getString(0);
+                            Log.d("CheckEmbedded", "file url: " + fileUrl);
+                            try {
+                                mmr.setDataSource(fileUrl);
+                                byte[] image = mmr.getEmbeddedPicture();
+                                if (image != null) {
+                                    db.execSQL("UPDATE " + PodDBAdapter.TABLE_NAME_FEED_MEDIA
+                                            + " SET " + PodDBAdapter.KEY_HAS_EMBEDDED_PICTURE + "=1"
+                                            + " WHERE " + PodDBAdapter.KEY_FILE_URL + "='"+ fileUrl + "'");
+                                } else {
+                                    db.execSQL("UPDATE " + PodDBAdapter.TABLE_NAME_FEED_MEDIA
+                                            + " SET " + PodDBAdapter.KEY_HAS_EMBEDDED_PICTURE + "=0"
+                                            + " WHERE " + PodDBAdapter.KEY_FILE_URL + "='"+ fileUrl + "'");
+                                }
+                            } catch(Exception e) {
+                                e.printStackTrace();
+                            }
+                        } while(c.moveToNext());
+                    }
+                }
+            });
+            t.start();
```

Comment:
Is anything waiting for this to complete? I don't see it. Am I missing something?

If not it would seem that the upgrade starts, the app goes on and does stuff while the DB is being modified.  Am I wrong?